src/Infrastructure/Service/Setting/ProjectSettingsInitializer/ProjectSettingsInitializer.php

```diff
@@ -0,0 +1,186 @@
+    {
+        $settingsToSave = [];
+
+        foreach ($settings as $setting) {
+            $settingToSave = $this->handleProjectSetting($setting, $projectSettingsDto);
+
+            if ($settingToSave === null) {
+                continue;
+            }
+
+            $settingsToSave[] = $settingToSave;
+        }
+
+        return $settingsToSave;
+    }
+
+    /**
+     * @param \SprykerSdk\SdkContracts\Entity\SettingInterface $setting
+     * @param \SprykerSdk\Sdk\Core\Application\Dto\ProjectSettingsInitDto $projectSettingsDto
+     *
+     * @return \SprykerSdk\SdkContracts\Entity\SettingInterface|null
+     */
+    protected function handleProjectSetting(SettingInterface $setting, ProjectSettingsInitDto $projectSettingsDto): ?SettingInterface
+    {
+        $values = $projectSettingsDto->getSettingValue($setting->getPath());
+
+        if ($projectSettingsDto->useDefaultValue() || $this->isInitializedByInitializer($setting)) {
+            return null;
+        }
+
+        if ($this->isValuesEmpty($values)) {
+            if (!$this->changeDefaultValueQuestion->ask($setting)) {
+                return null;
+            }
+
+            $values = $this->settingValueQuestion->ask($setting, $setting->getValues());
+        }
+
+        if (!$this->isValuesChanged($setting, $values)) {
+            return null;
+        }
+
+        $setting->setValues($values);
+
+        return $setting;
+    }
+
+    /**
+     * @param mixed $values
+     *
+     * @return bool
+     */
+    protected function isValuesEmpty($values): bool
+    {
+        return $values === null || $values === [];
+    }
+
+    /**
+     * @param array<\SprykerSdk\SdkContracts\Entity\SettingInterface> $settings
+     *
+     * @return void
+     */
+    protected function initializeSettings(array $settings): void
+    {
+        foreach ($settings as $setting) {
+            $initializerName = $setting->getInitializer();
+
+            if ($initializerName === null || !$this->settingInitializerRegistry->hasSettingInitializer($initializerName)) {
+                continue;
+            }
+
+            $this->settingInitializerRegistry->getSettingInitializer($initializerName)->initialize($setting);
+        }
+    }
+
+    /**
+     * @param \SprykerSdk\SdkContracts\Entity\SettingInterface $setting
+     *
+     * @return bool
+     */
+    protected function isInitializedByInitializer(SettingInterface $setting): bool
+    {
+        return !$setting->hasInitialization()
+            || ($setting->getInitializer() && $setting->getType() === ValueTypeEnum::TYPE_UUID);
+    }
+
+    /**
+     * @param \SprykerSdk\SdkContracts\Entity\SettingInterface $setting
+     * @param mixed $values
+     *
+     * @return bool
+     */
+    protected function isValuesChanged(SettingInterface $setting, $values): bool
+    {
+        $values = [ValueTypeEnum::TYPE_BOOLEAN => (bool)$values, ValueTypeEnum::TYPE_ARRAY => (array)$values][$setting->getType()] ?? (string)$values;
```

Comment:
1. Please, split the line it's longer than 120 characters.
2. Avoid using complex inline expressions to increase readability.